### ISSUE TITLE
Standardise testing semantics

### ISSUE THREAD

**Issue by ellsclytn** [everydayhero/react-widgets#370]
Tests throughout the code seem to vary in how they are described, including both the `describe` and `it` statements. 

Multiple recommendations exist regarding how these should be phrased, so we need to decide on a standard to adhere to.

**Comment by ellsclytn** [comment]
My own suggestion is that `describe` should announce what is being tested, in plain English (not necessarily the exact property/variable name).

Plain English can serve to provide a clear context of what is being tested, whereas the use of a property/variable name to describe a test can be considerably more vague, especially when variables are getting reassigned/renamed between `export`s or callbacks.

For example, `describe('Displaying an icon'` gives a clear idea of what is being tested, but doesn't go as in depth to describe variable names, which can be confusing if there's multiple variables involved (eg `renderIcon` & `iconName`).

`it` declarations can instead be as specific to describe the key variables/properties involved (or explicitly *not* involved, along with their expected function, eg:

- `it('renders no icon when renderIcon set to false'`
- `it('defaults to render a short format when format is not specified'`

**Comment by slwen** [comment]
I agree it would be nice to see some consistency in formatting/wording :+1: Perhaps something for a `CONTRIBUTING.md`

**Comment by ellsclytn** [comment]
Solution for V4 will be:

- Have all tests migrated across from Jest to Mocha
- Produce a `CONTRIBUTING.md` document that described the semantics behind unit tests.
